### PR TITLE
feat(carla): add CARLA integration

### PR DESCRIPTION
**Pull request by zebin-huang**
Description:
- Integrated CARLA for simulation and testing.
- Updated documentation with setup and usage instructions.

### CODE REVIEW

File: carla/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+### Testing
+
+```bash
+# Run CARLA in the installation path, run:
+./CarlaUE4.sh
+
+# Evaluate the model with the following command:
+python evaluate_model_instructions.py --episode_file ./datasets/translated_test_suites/Town02_All.txt --model "ModifiedDeepestLSTMTinyPilotNet/models/v10.0.pth" --n_episodes 100 --combined_control --bert_model_path ./models/checkpoint-1000 --tokenizer_name huawei-noah/TinyBERT_General_4L_312D --label_mapping_path ./models/label_mapping.json
```
Comment (sergiopaniego) [inline carla/README.md]: Do we have instructions for downloading the `checkpoint-1000` model that is used as input for `bert_model_path`? If not, maybe we could add them 😄

Comment (zebin-huang) [reply carla/README.md]: I have updated this part, and the final command would be --combined_control --bert_model_path ./models/tinybert_model.pt.

Comment (zebin-huang) [reply carla/README.md]: The interface has been changed due to the adoption of a new model loading method, and the model is already available on Hugging Face.

**Comment by sergiopaniego** [comment]
You may change the PR to `Ready for review` when completed :smile:

**Pull request merged**